mothership-ec/cog
# Lazy Loading

#### What does this do?

Adds central `EntityLoaderInterface` and `EntityLoaderCollection` for lazy loading entities in the system.
#### How should this be manually tested?

Go to commerce and test PR https://github.com/messagedigital/cog-mothership-commerce/pull/367, to see the interface and collection in action.
Go through the changes in the readme and see whether everything important is in there.
#### Related PRs / Issues / Resources?

Lazy loading for products: https://github.com/messagedigital/cog-mothership-commerce/pull/367
Lazy loading for file tags: https://github.com/messagedigital/cog-mothership-file-manager/pull/72
Lazy loading for page tags: https://github.com/messagedigital/cog-mothership-cms/pull/216
#### Anything else to add? (Screenshots, background context, etc)
- [x] Add `EntityLoaderInterface` in Cog
- [x] Add `EntityLoaderCollection` in Cog
- [x] Update documentation (readme) in Cog, explain functionality, naming conventions and how to use it
